content/en/tracing/trace_collection/custom_instrumentation/java/otel.md

```diff
@@ -17,118 +17,205 @@ further_reading:
 
-### Add custom spans using annotations
+If you have existing @Trace or similar annotations, or prefer to use annotations to complete any incomplete traces within Datadog, use Trace Annotations.
 
-First add a dependency to the `opentelemetry-instrumentation-annotations` library.
+Traces may also be created using the OpenTelemetrey `@WithSpan` annotation as described in [Trace annotations](#trace-annotations).
 
-{{< tabs >}}
-{{% tab "Gradle" %}}
+## Trace annotations
 
-{{< code-block lang="groovy" >}}
-  // OpenTelemetry instrumentation annotations
-  implementation "io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:${opentelemetryVersion}"
-{{< /code-block >}}
+Add `@WithSpan` to methods to have them be traced when running OpenTelemetry and the `dd-java-agent.jar`. If the Agent is not attached, this annotation has no effect on your application.
+OpenTelemetry's `@WithSpan` annotation is provided by the `opentelemetry-instrumentation-annotations` dependency.
 
-{{% /tab %}}
-{{% tab "Maven" %}}
+```java
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+
+public class SessionManager {
 
-{{< code-block lang="xml" >}}
-<!-- OpenTelemetry instrumentation annotations -->
-<dependency>
-    <groupId>io.opentelemetry.instrumentation</groupId>
-    <artifactId>opentelemetry-instrumentation-annotations</artifactId>
-    <version>${io.opentelemetry.version}</version>
-</dependency>
-{{< /code-block >}}
+  @WithSpan
+  public static void saveSession() {
+    // your method implementation here
+  }
+}
+```
 
-{{% /tab %}}
-{{< /tabs >}}
+## Creating new spans
 
-Then annotate your methods with the `@WithSpan` annotation to create a new span each call. The parameters of the call can be annotated with the `@SpanAttribute` annotation to capture the arguments as span attributes:
+To manually create new spans that start a new, independent trace:
 
 ```java
-@WithSpan
-public void myMethod(@SpanAttribute("parameter1") String parameter1,
-    @SpanAttribute("parameter2") long parameter2) {
-    <...>
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.ResourceAttributes;
+import java.util.concurrent.TimeUnit;
+
+public class Example {
+
+  public void doSomething() {
+    Tracer tracer = getOpenTelemetry().getTracer("my-scope", "0.1.0");
+    Span span = tracer.spanBuilder("my-resource").startSpan();
+    try (Scope scope = span.makeCurrent()) {
+      // do some work
+    } catch (Throwable t) {
+      span.recordException(t);
+      throw t;
+    } finally {
+      span.end();
+    }
```

Comment:
This code has nothing related with creating a new independant trace.
It will create a span (`my-resource`) that will be attached to the current trace.
Instead, you are defining the new span as current context for the local thread. 

If you want to create a new trace, use `setNoParent()`, or alternatively a root context as parent.

Reply:
Fixed the introductory sentence to better reflect what the example does. 👍

Reply:
Great! But wasn't the point of the sample to show how to create a new trace? 🤷  (real question, I don't know what the customer is asking or interested about)

Reply:
The main goal for this section is **Manually creating a new span**.